src/utils/detectors.js

```diff
@@ -14,8 +14,10 @@ export const isValidTopLevelImport = (x, state) =>
 
 const localNameCache = {}
 
-export const importLocalName = (name, state, bypassCache = false) => {
-  const cacheKey = name + state.file.opts.filename
+export const importLocalName = (name, state, options = {}) => {
+  const { cacheIdentifier, bypassCache = false } = options;
+  const cacheKeyAffix = cacheIdentifier ? `|${cacheIdentifier}` : '';
+  const cacheKey = name + state.file.opts.filename + cacheKeyAffix;
```

Comment:
I'm not married to the idea of affixing the value, so long as it's included in the `cacheKey`